### ISSUE TITLE
"file not included in module tree" is reported after changing the case of filename

### ISSUE THREAD

**Issue by WOo0W** [rust-analyzer/rust-analyzer#9947]
I'm using VSCode with `rust-analyzer 0.2.710` on Windows 10.

To reproduce:

1. Create a new rust project
2. Create `A.rs` under `src`
3. Rename `A.rs` to `a.rs`
4. Add `mod a;` to `main.rs`
5. In `a.rs` it reports `file not included in module tree` and lints are disabled.

![image](https://user-images.githubusercontent.com/24245520/130006872-29b5964f-da10-4eb5-b73d-8bb9a171cb97.png)

![image](https://user-images.githubusercontent.com/24245520/130006859-d7f02f2c-8c6e-4ea0-b3b0-ae40261fabbd.png)

**Comment by Veykril** [comment]
Can confirm this happening on latest nightly on windows 10 as well(including restarts not fixing this).
Note that the quickfix there wants to insert a `mod A;` so RA thinks the file is still called `A.rs`. I assume this is windows' case insensitivity biting us somehow...

**Comment by Veykril** [comment]
So I think this is a windows problem, that is it doesn't actually rename the file so we get the first version of the path back no matter what.

You can fix this yourself by renaming the file to something else(a rename that doesnt only change casing), then change it back to what you desire, that should fix the path.

**Comment by WOo0W** [comment]
> You can fix this yourself by renaming the file to something else(a rename that doesnt only change casing), then change it back to what you desire, that should fix the path.

This doesn't work. Even delete the file and recreate it.
Maybe this is cached by `rust-analyzer`. How can I clear the cache?

**Comment by Veykril** [comment]
rust-analyzer does not cache anything. If you restart the client/server its wiped. So this is most likely a VSCode issue, would be good to confirm if someone using another client could try reproducing this behaviour on windows.

> This doesn't work. Even delete the file and recreate it.

For me when I create a file according to your scheme, then rename `a.rs` to `b.rs` then back to `a.rs` fixed it for me...

**Comment by WOo0W** [comment]
> For me when I create a file according to your scheme, then rename a.rs to b.rs then back to a.rs fixed it for me...

It doesn't really fix it. Just the tip disappears at first. Try to type and you will find the IntelliSense unaviliable. Type something and it will be back.

**Comment by Veykril** [comment]
Ah you are right, I didn't go that far.

Can confirm you can fix it by doing what this comment proposes: https://github.com/microsoft/vscode/issues/12448#issuecomment-655001292

So ye this is a caching issue with vscode on windows, not a RA specific issue.

**Comment by Veykril** [comment]
So looking further into this it looks like this is a `wont-fix` https://github.com/microsoft/vscode/issues/102627

**Comment by WOo0W** [comment]
Deleting the `state.vscdb` fixes it.

**Comment by Veykril** [comment]
Let's keep this open for other people to find then as I believe this to be the second time this got reported(and now we know whats up with it).

I edited the comment link into the issue itself for visibility.

**Comment by Hollow-Glitch** [comment]
Hi there I have encountered this very problem, and before finding issue, I described my problem here: https://github.com/rust-lang/rust-analyzer/issues/12697#issuecomment-1445329882.
I can confirm that what @Veykril  is pointing to, i.e.: https://github.com/microsoft/vscode/issues/12448#issuecomment-655001292
indeed is the solution/**workaround** to this problem, and removing `state.vscdb` in the workspace's "AppData" folder solved it on my side.

Found the following extension that could simplify if this is occurring to someone often: https://marketplace.visualstudio.com/items?itemName=MamoruDS.workspace-cacheclean

I would like to state that I am **unsatisfied** with this state of simple file-renames causing problems, hopefully none of us are!

**Comment by ChainThemAll** [comment]
This problem seems to be unsolvable in Mac?

**Comment by belchior** [comment]
To fix in MacOS I uninstall the extension and removed the folder of the rust-analyzer

```sh
cd
rm -rf ./.vscode/extensions/rust-lang.rust-analyzer-0.3.2180-darwin-x64 
```

**Comment by Andrewsimsd** [comment]
> Deleting the `state.vscdb` fixes it.

state.vscdb is in %APPDATA%/Code/User/globalStorage/

**Comment by ShadowNineX** [comment]
GEEEEEZZZZZZ 2021????? Can somebody fix this please. EDIT: Damn. Microsoft did a wontfix? What a shame.
